[CI] StringTemplatePluginTest test substitution failing

Also failing

StringTemplatePluginTest » test basic conditional	
StringTemplatePluginTest » test if then else
StringTemplatePluginTest » output file already present and up to date
StringTemplatePluginTest » test substitution
StringTemplatePluginTest » output file already present but not up to date

Failures seem to be restricted to Windows builds.

Root cause appears to be related to syntax errors in generated gradle build files, e.g.

* What went wrong:	
Could not compile build file 'C:\Users\jenkins\AppData\Local\Temp\junit4118947683707987252\build.gradle'.	
> startup failed:	
  build file 'C:\Users\jenkins\AppData\Local\Temp\junit4118947683707987252\build.gradle': 28: Unexpected character: '"' @ line 28, column 39.	
                   it.inputFile = new File("C:\Users\jenkins\AppData\Local\Temp\junit4118947683707987252/src/main/Color.txt.st")

**Build scan:**
https://gradle-enterprise.elastic.co/s/hc3sn2lrswuy4/tests/:integTest/org.elasticsearch.gradle.internal.StringTemplatePluginTest/test%20substitution

**Reproduction line:**
```
null
```

**Applicable branches:**
main

**Reproduces locally?:**
No

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.gradle.internal.StringTemplatePluginTest&tests.test=test%20substitution

**Failure excerpt:**
```
org.gradle.testkit.runner.UnexpectedBuildFailure: Unexpected build execution failure in C:\Users\jenkins\AppData\Local\Temp\junit1828736541946069288 with arguments [stringTemplates, -g, C:\Users\jenkins\AppData\Local\Temp\spock_test_substitution_0_gradleUserHome10067918894284790678, --configuration-cache, -Dtest.external=true]

Output:
Calculating task graph as no configuration cache is available for tasks: stringTemplates

FAILURE: Build failed with an exception.

* Where:
Build file 'C:\Users\jenkins\AppData\Local\Temp\junit1828736541946069288\build.gradle' line: 28

* What went wrong:
Could not compile build file 'C:\Users\jenkins\AppData\Local\Temp\junit1828736541946069288\build.gradle'.
> startup failed:
  build file 'C:\Users\jenkins\AppData\Local\Temp\junit1828736541946069288\build.gradle': 28: Unexpected character: '"' @ line 28, column 39.
                   it.inputFile = new File("C:\Users\jenkins\AppData\Local\Temp\junit1828736541946069288/src/main/p/X-Box.java.st")
                                           ^
  
  1 error


* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

BUILD FAILED in 1s
Configuration cache entry stored.

  at org.gradle.testkit.runner.internal.DefaultGradleRunner.lambda$build$2(DefaultGradleRunner.java:269)
  at org.gradle.testkit.runner.internal.DefaultGradleRunner.run(DefaultGradleRunner.java:362)
  at org.gradle.testkit.runner.internal.DefaultGradleRunner.build(DefaultGradleRunner.java:267)
  at org.elasticsearch.gradle.internal.test.InternalAwareGradleRunner.build(InternalAwareGradleRunner.java:145)
  at org.elasticsearch.gradle.internal.test.ConfigurationCacheCompatibleAwareGradleRunner.build(ConfigurationCacheCompatibleAwareGradleRunner.java:153)
  at org.elasticsearch.gradle.internal.test.NormalizeOutputGradleRunner.build(NormalizeOutputGradleRunner.java:147)
  at org.elasticsearch.gradle.internal.StringTemplatePluginTest.test substitution(StringTemplatePluginTest.groovy:50)

```

--- a/build-tools-internal/src/integTest/groovy/org/elasticsearch/gradle/internal/StringTemplatePluginTest.groovy
+++ b/build-tools-internal/src/integTest/groovy/org/elasticsearch/gradle/internal/StringTemplatePluginTest.groovy
@@ -35,12 +35,12 @@ class StringTemplatePluginTest extends AbstractGradleFuncTest {
           tasks.named("stringTemplates").configure {
             template {
               it.properties = ["Type" : "Int", "type" : "int"]
-              it.inputFile = new File(projectDir, "src/main/p/X-Box.java.st")
+              it.inputFile = file("src/main/p/X-Box.java.st")
               it.outputFile = "p/IntBox.java"
             }
             template {
               it.properties = ["Type" : "Long", "type" : "long"]
-              it.inputFile = new File(projectDir, "src/main/p/X-Box.java.st")
+              it.inputFile = file("src/main/p/X-Box.java.st")
               it.outputFile = "p/LongBox.java"
             }
           }
@@ -90,7 +90,7 @@ class StringTemplatePluginTest extends AbstractGradleFuncTest {
           tasks.named("stringTemplates").configure {
             template {
               it.properties = ["Red" : "true", "Blue" : "", "Green" : "true"]
-              it.inputFile = new File(projectDir, "src/main/Color.txt.st")
+              it.inputFile = file("src/main/Color.txt.st")
               it.outputFile = "Color.txt"
             }
           }
@@ -126,7 +126,7 @@ class StringTemplatePluginTest extends AbstractGradleFuncTest {
           tasks.named("stringTemplates").configure {
             template {
               it.properties = [:] // no properties
-              it.inputFile = new File(projectDir, "src/main/Token.txt.st")
+              it.inputFile = file("src/main/Token.txt.st")
               it.outputFile = "Token.txt"
             }
           }
@@ -162,7 +162,7 @@ class StringTemplatePluginTest extends AbstractGradleFuncTest {
           tasks.named("stringTemplates").configure {
             template {
               it.properties = [:] // no properties
-              it.inputFile = new File(projectDir, "src/main/UpToDate.txt.st")
+              it.inputFile = file("src/main/UpToDate.txt.st")
               it.outputFile = "UpToDate.txt"
             }
           }
@@ -198,7 +198,7 @@ class StringTemplatePluginTest extends AbstractGradleFuncTest {
           tasks.named("stringTemplates").configure {
             template {
               it.properties = [:] // no properties
-              it.inputFile = new File(projectDir, "src/main/Message.txt.st")
+              it.inputFile = file("src/main/Message.txt.st")
               it.outputFile = "Message.txt"
             }
           }

--- a/build-tools-internal/src/integTest/groovy/org/elasticsearch/gradle/internal/StringTemplatePluginTest.groovy
+++ b/build-tools-internal/src/integTest/groovy/org/elasticsearch/gradle/internal/StringTemplatePluginTest.groovy
@@ -35,12 +35,12 @@ class StringTemplatePluginTest extends AbstractGradleFuncTest {
           tasks.named("stringTemplates").configure {
             template {
               it.properties = ["Type" : "Int", "type" : "int"]
-              it.inputFile = new File("${projectDir}/src/main/p/X-Box.java.st")
+              it.inputFile = new File(projectDir, "src/main/p/X-Box.java.st")
               it.outputFile = "p/IntBox.java"
             }
             template {
               it.properties = ["Type" : "Long", "type" : "long"]
-              it.inputFile = new File("${projectDir}/src/main/p/X-Box.java.st")
+              it.inputFile = new File(projectDir, "src/main/p/X-Box.java.st")
               it.outputFile = "p/LongBox.java"
             }
           }
@@ -90,7 +90,7 @@ class StringTemplatePluginTest extends AbstractGradleFuncTest {
           tasks.named("stringTemplates").configure {
             template {
               it.properties = ["Red" : "true", "Blue" : "", "Green" : "true"]
-              it.inputFile = new File("${projectDir}/src/main/Color.txt.st")
+              it.inputFile = new File(projectDir, "src/main/Color.txt.st")
               it.outputFile = "Color.txt"
             }
           }
@@ -126,7 +126,7 @@ class StringTemplatePluginTest extends AbstractGradleFuncTest {
           tasks.named("stringTemplates").configure {
             template {
               it.properties = [:] // no properties
-              it.inputFile = new File("${projectDir}/src/main/Token.txt.st")
+              it.inputFile = new File(projectDir, "src/main/Token.txt.st")
               it.outputFile = "Token.txt"
             }
           }
@@ -162,7 +162,7 @@ class StringTemplatePluginTest extends AbstractGradleFuncTest {
           tasks.named("stringTemplates").configure {
             template {
               it.properties = [:] // no properties
-              it.inputFile = new File("${projectDir}/src/main/UpToDate.txt.st")
+              it.inputFile = new File(projectDir, "src/main/UpToDate.txt.st")
               it.outputFile = "UpToDate.txt"
             }
           }
@@ -198,7 +198,7 @@ class StringTemplatePluginTest extends AbstractGradleFuncTest {
           tasks.named("stringTemplates").configure {
             template {
               it.properties = [:] // no properties
-              it.inputFile = new File("${projectDir}/src/main/Message.txt.st")
+              it.inputFile = new File(projectDir, "src/main/Message.txt.st")
               it.outputFile = "Message.txt"
             }
           }

